### PR TITLE
Add flox channel update automation

### LINKED ISSUE #2: Automate periodic pulls from upstream
Up until now we have been reconciling updates from upstream using a script on account of the requirement to address merge conflicts, but would like to set up a GitHub-based automation to handle this going forward. General requirements for the automation:

* Make automation configurable with regard to the enumeration of stabilities, merge frequencies, dates, etc. It's OK to expose code as a configuration mechanism.
* Perform following steps (in this order) daily at 12pm (noon) GMT:
    * Every day, fetch updates from upstream.
    * _Before_ updating a branch, create tag of the form `<branchname>.<YYYYMMDD>`, e.g. `unstable.20220420` pointing to the current (soon to be replaced) HEAD.
    * On Monday thru Wednesday, rebase updates from `nixos-unstable` upstream into the `unstable` branch.
    * On the first Saturday of the month (or configurable), reset the `stable` branch to point to the current `staging`.
    * On Saturday, reset the `staging` branch to point to the current `unstable`.

@samrose - I'll provide you a copy of the current script offline.

A graphical representation of the workflow is shown below:
image

### PR DESCRIPTION
**Pull request by ysndr**
closes #2 

- had to use a new PAT that contains the `workflows` permission (as the action is contained in the rebase onto unstable). @notgne2 Can we update the current CI token with that permission? (Sorry if you are not the right one to ask, I figured you may be the ops guy lately)

- Uses the script @limeytexan was using before with minimal changes.
- runs on a schedule every day at 8am UTC, @limeytexan tell me if that's too early/late.
- will create a comment on an issue we can dedicate for them. (or create issues on its own, you tell me)
- annotates the action run with notes telling what was updated

See a full run [here](https://github.com/flox/not-nixpkgs/actions/runs/3234435388)

If we ever provision such a channel management to people, we should refactor the steps and stabilities, but for now (and the last 2y as I was told) the script is good enough™️

### CODE REVIEW

File: .github/workflows/flox-channels.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    if: github.repository_owner == 'flox'
+    runs-on: ubuntu-latest
+
+    # TODO: implement configurable merge based on matrix
+
+    name: "Periodic pull from upstream"
+    steps:
+      - name: Configure Git Env
+        run: |
+          git config --global user.email "automations@floxdev.com"
```
Comment (limeytexan) [inline .github/workflows/flox-channels.yml]: We have a "technology@floxdev.com" group alias but not an "automations" one ... are you wanting to create that or should we just use technology?

File: .github/workflows/flox-channels.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          git config --global user.name "Flox Nixpkgs Updater"
+
```
Comment (limeytexan) [inline .github/workflows/flox-channels.yml:52]:
```suggestion

        # gh automatically sets the `upstream` remote to github:NixOS/nixpkgs
```

**Review by limeytexan — CHANGES_REQUESTED** [review]
Looks good, thanks. I wish we could use the same clone upon successive iterations to avoid all the unnecessary cloning, but this should do for now.

Discussed verbally that we should also do the "day of week" check in the "configure git env" step to avoid unnecessary cloning when we're not going to do anything anyway.

**Comment by ysndr** [comment]
Day of week is verified in a separate job. Teh update job is run only if the previous one sets a guard variable.

**Review by tomberek — CHANGES_REQUESTED** [review]
I would much rather have this automation live in nixpkgs-flox than as a constant rebase in flox/nixpkgs.

**Comment by ysndr** [comment]
Closing in favor of https://github.com/flox/nixpkgs-flox/pull/16

**Pull request closed without merging**